src/rocker/nvidia_extension.py

```diff
@@ -42,18 +40,18 @@ def get_name():
     def __init__(self):
         self.name = X11.get_name()
         self._env_subs = None
-        self._xauth = tempfile.NamedTemporaryFile(prefix='.docker', suffix='.xauth')
```

Comment:
There is probably a reason behind this?

Reply:
This was made unique to keep multiple containers from interfering with each other.

The tempfile is the safe way to make a file like this to ensure that there's no collisions etc.

What I would recommend instead is to extend the nocleanup logic to make the tempfile persist. 

An extension would be to also think about how to have a second step which would clean up any dangling resources like this for a container previously passed nocleanup